### PR TITLE
feat: [CDS-73414]: Enable manifest fetch task and fix chart.yaml directory

### PR DESCRIPTION
**Pull request by abosii-harness**
## Describe your changes
This PR includes next changes:
- Register beans in kryo
- Fix an issue when if an error is thrown in the cache lambda it was wrapped into unchecked exception and messed up with overall error experience
- Fix an issue when using helm repositories the output path contains the helm chart name 
- Enable  HELM_FETCH_CHART_MANIFEST_TASK

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49992)

### CODE REVIEW

**Comment by abosii-harness** [comment]
trigger checkstylecodeformat, commonchecks, uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/base/HelmChartManifestTaskService.java [hunk 1]
```diff
@@ -164,8 +165,9 @@ private void fetchHelmChartYamlFromGit(
     }
 
     String chartYamlOutputPath = getChartYamlPath(manifestConfig, destinationDirectory);
-    FileIo.createDirectoryIfDoesNotExist(Paths.get(chartYamlOutputPath).getParent());
-    FileIo.waitForDirectoryToBeAccessibleOutOfProcess(chartYamlPath, 10);
+    Path chartYamlDirectory = Paths.get(chartYamlOutputPath).getParent();
+    FileIo.createDirectoryIfDoesNotExist(chartYamlDirectory);
+    FileIo.waitForDirectoryToBeAccessibleOutOfProcess(chartYamlDirectory.toString(), 10);
```
Comment (acasian-harness) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/base/HelmChartManifestTaskService.java:170]: this value of 10 - is it seconds or ms?

**Review by acasian-harness — DISMISSED** [review]

**Review by GoelPratyush99 — DISMISSED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/base/HelmChartManifestTaskService.java (hunk 1 above)
Comment (TarunUba1) [reply 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/base/HelmChartManifestTaskService.java:170]: they are max retries

**Review by TarunUba1 — DISMISSED** [review]

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/base/HelmChartManifestTaskService.java [hunk 2]
```diff
@@ -193,6 +195,25 @@ private void fetchHelmChartYamlFromHelmRepo(HelmChartManifestDelegateConfig mani
     }
   }
 
+  private File findHelmChartYaml(HelmChartManifestDelegateConfig manifestConfig, String destinationDirectory) {
+    String chartYamlSubpath;
+    if (isNotEmpty(manifestConfig.getSubChartPath())) {
+      chartYamlSubpath = Paths.get(manifestConfig.getSubChartPath(), CHARTS_YAML_KEY).toString();
+    } else {
+      chartYamlSubpath = CHARTS_YAML_KEY;
+    }
```
Comment (abhinavsingh1196) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/helm/base/HelmChartManifestTaskService.java:204]: nit: the above logic in another method would make this more readable? 
`String chartYamlSubpath = findSubPath(..)`

**Review by abhinavsingh1196 — APPROVED** [review]

**Review by TarunUba1 — APPROVED** [review]

**Comment by abosii-harness** [comment]
trigger review, messagecheck

**Review by namantalaycha10 — APPROVED** [review]

**Pull request merged**